Controllers/OperationController.cs

```diff
@@ -16,18 +17,45 @@ namespace ARMExplorer.Controllers
     public class OperationController : ApiController
     {
         private readonly IArmRepository _armRepository;
+        private static readonly MemoryCache SwaggerCache = new MemoryCache("SwaggerDefinitionCache");
```

Comment:
yes. we could use ConcurrentDictionary and achieve the same results. but i felt MemoryCache made the intent clearer and satisfied our thread safety requirements.

I can add the following comment regarding the Lazy pattern
AddOrGetExisting covers a narrow case where 2 calls come in at the same time for the same provider then its swagger will be parsed twice. the Lazy pattern guarantees each swagger will ever be parsed only once and other concurrent accesses for the same providerkey will be blocked until the previous thread adds the value to cache.